python/graphstorm/gconstruct/utils.py

```diff
@@ -530,6 +538,91 @@ def to_tensor(self):
         """
         return th.tensor(self.to_numpy())
 
+class ExtFeatureWrapper(ExtNumpyWrapper):
+    """ The wrapper to memory-mapped Numpy array when combining feature
+
+    Parameters
+    ----------
+    arr_path : str
+        The path to different feature files directory
+    merged_file: str
+        The merged file name
+    wrapper: list
+        List of ExtNumpyWrapper
+    """
+    def __init__(self, arr_path, shape=None, dtype=None, merged_file="merged_feature.npy"):
+        super().__init__(arr_path, shape, dtype)
+        self.directory_path = arr_path
+        self.merged_file = merged_file
+        self.wrapper = []
+        self._arr_path = os.path.join(self.directory_path, self.merged_file)
+
+    def __getitem__(self, idx):
+        if not self._shape:
+            raise RuntimeError("The ExtFeatureWrapper needs to be merge first")
+        if self._arr is None:
+            self._arr = np.memmap(self._arr_path, self._dtype, mode="r",
+                                  shape=self._shape)
+        return self._arr[idx].astype(self.dtype)
+
+    def __del__(self):
+        self.cleanup()
+
+    def cleanup(self):
+        """ Clean up the array.
+        """
+        self._arr.flush()
+        self._arr = None
+        shutil.rmtree(self.directory_path)
+
+    def to_numpy(self):
+        """ Convert the data to Numpy array.
+        """
+        if not self._shape:
+            raise RuntimeError("The ExtFeatureWrapper needs to be merge first")
+        if self._arr is None:
+            arr = np.memmap(self._arr_path, self._orig_dtype, mode="r", shape=self._shape)
+            if self._dtype != self._orig_dtype:
+                arr = arr.astype(self._dtype)
+            return arr
+        else:
+            return self._arr.astype(self._dtype)
+
+    def append(self, wrap):
+        """Add external memory wrapper to the wrapper
+        """
+        if not self.shape:
+            self._shape = (wrap.shape[0], 0)
+        if self.shape and self.shape[0] != wrap.shape[0]:
+            raise RuntimeError("Expect that ExtFeatureWrapper has a "
+                               "first dimension that is the same")
```

Comment:
Expect that the first dimension of the ExtFeatureWrapper and wrap is same but get xxx and xxx.